Tools/peg_generator/scripts/test_parse_directory.py

```diff
@@ -185,27 +155,38 @@ def parse_directory(
             if PurePath(file).match(pattern):
                 should_exclude_file = True
                 break
+        with open(file, "rb") as f:
+            source = f.read()
 
         if not should_exclude_file:
             try:
                 if tree_arg:
                     mode = 1
-                if parser == "cpython":
-                    with open(file, "r") as f:
-                        source = f.read()
-                        if mode == 2:
-                            compile(source, file, "exec")
-                        elif mode == 1:
-                            ast.parse(source, file, "exec")
+                if parser == "pegen":
```

Comment:
Isn't this backward?

Reply:
I don't think so, since the new parser is the default.

Reply:
Okay, but it looks weird, and it might not be the default if `-X oldparser` is used. Could you use the `_peg_parser` functions with `old_parser=False`?